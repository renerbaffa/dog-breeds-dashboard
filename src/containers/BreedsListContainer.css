.container {
  height: 100%;
}

.title {
  display: flex;
  justify-content: center;
  padding: 8px;
  font-weight: bold;
  font-size: 20px;
}
