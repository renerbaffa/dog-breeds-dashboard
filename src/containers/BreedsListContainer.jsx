import React, { Component } from 'react';

import BreedsList from '../components/breeds/BreedsList'; // eslint-disable-line
import BreedSearchForm from '../components/breeds/BreedSearchForm';

import styles from './BreedsListContainer.css';

class BreedsListContainer extends Component {
  state = {
    searchText: '',
  }

  handleSearchTextChange = event =>
    this.setState({ searchText: event.target.value });

  render() {
    const { searchText } = this.state;

    return (
      <div className={styles.container}>
        <BreedSearchForm searchText={searchText} />
        <BreedsList searchText={searchText} />
      </div>
    );
  }
}

export default BreedsListContainer;
