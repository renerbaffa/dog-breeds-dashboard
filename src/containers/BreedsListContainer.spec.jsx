import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import BreedsListContainer from './BreedsListContainer';

jest.mock('../components/breeds/BreedsList', () => 'BreedsList');

describe('<BreedsListContainer />', () => {
  let component;
  let wrapper;

  beforeEach(() => {
    component = <BreedsListContainer />;
    wrapper = shallow(component);
  });

  it('should render correctly', () => {
    const container = renderer.create(component).toJSON();
    expect(container).toMatchSnapshot();
  });

  it('should set empty string as default state', () => {
    expect(wrapper.state().searchText).toBe('');
  });

  it('should update searchText state', () => {
    const newValue = 'new value';
    expect(wrapper.state().searchText).toBe('');
    wrapper
      .instance()
      .handleSearchTextChange({ target: { value: newValue } });
    expect(wrapper.state().searchText).toBe(newValue);
  });
});
