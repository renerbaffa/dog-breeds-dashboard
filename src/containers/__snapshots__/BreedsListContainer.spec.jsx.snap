// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BreedsListContainer /> should render correctly 1`] = `
<div
  className="container"
>
  <div
    className="title"
  >
    Dog breeds
  </div>
  <div
    className="container"
  >
    <form
      className="form"
      onSubmit={[Function]}
    >
      <input
        className="input"
        onChange={[Function]}
        placeholder="Search..."
        type="text"
        value=""
      />
      <button
        className="button main"
      >
        <svg
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": undefined,
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em"
        >
          <g>
            <path
              d="m15.9 23.4c4.1 0 7.5-3.4 7.5-7.5s-3.4-7.5-7.5-7.5-7.5 3.3-7.5 7.5 3.3 7.5 7.5 7.5z m10 0l8.2 8.2-2.5 2.5-8.2-8.2v-1.4l-0.5-0.4c-1.9 1.6-4.4 2.5-7 2.5-6.1 0-10.9-4.7-10.9-10.7s4.8-10.9 10.9-10.9 10.7 4.8 10.7 10.9c0 2.6-0.9 5.1-2.5 7l0.4 0.5h1.4z"
            />
          </g>
        </svg>
      </button>
    </form>
  </div>
  <BreedsList
    searchText=""
  />
</div>
`;
